### PR TITLE
Support additional types in csv and fixes

### PR DESCRIPTION
**Pull request by markaren**
* Support additional FMI types in CSV. Declared in header. Defaults to `real`.
* Fix index stepper logic

### CODE REVIEW

**Comment by markaren** [comment]
@fcollonval Have a look if you want/have time in case there are obvious things that could be done more pythonic.

File: pythonfmu/csvbuilder.py [hunk 1]
```diff
@@ -23,9 +25,48 @@ def lerp(v0: float, v1: float, t: float) -> float:
...
+    Fmi2Type.integer: Integer,
+    Fmi2Type.real: Real,
+    Fmi2Type.boolean: Boolean,
+    Fmi2Type.string: String
+    }}
+
+class Header:
+
+    def __init__(self, s):
+        m = re.search(r"\\[(.*?)\\]", s)
```
Comment (markaren) [inline pythonfmu/csvbuilder.py]: This regex is broad by choice.

File: pythonfmu/csvbuilder.py [hunk 2]
```diff
@@ -1,30 +1,68 @@
...
+  elif Fmi2Type.real.name in s_lower:
+    return Fmi2Type.real
+  elif Fmi2Type.boolean.name in s_lower:
+    return Fmi2Type.boolean
+  elif Fmi2Type.string.name in s_lower:
+    return Fmi2Type.string
+  elif Fmi2Type.enumeration.name in s_lower:
+    raise TypeError("Unsupported type: {{Fmi2Type.enumeration.name}}")
+  else:
+    raise TypeError("Could not process type from input string: {{s}}")
```
Comment (fcollonval) [inline pythonfmu/csvbuilder.py]: What about this version? Note: I think you should be more stringent how the format to provide the type. Checking for `type in` is too broad. It is better to force the type to `startswith` or `endswith`.

```suggestion
def get_fmi2_type(s: str) -> Fmi2Type:
    s_lower = s.lower()
    for type in Fmi2Type:
        if type.name in s_lower:
            if type == Fmi2Type.enumeration:
                raise TypeError("Unsupported type: {{Fmi2Type.enumeration.name}}")
            else:
                return type
    raise TypeError("Could not process type from input string: {{s}}")
```

File: pythonfmu/tests/csvdemo.csv [hunk 3]
```diff
@@ -1,7 +1,7 @@
-t, out1, out2
-0.0, 1, 2
-0.1, 2, 4
-0.2, 3, 8
-0.3, 4, 16
-0.4, 5, 32
-0.5, 6, 64
+t, int [1 INTEGER], real[REAL], bool[BOOLEAN], string [STRING]
```
[the remark this answers is not in the document]
Comment (markaren) [reply pythonfmu/tests/csvdemo.csv]: Good Idea. I can take the last group match.

File: pythonfmu/csvbuilder.py (hunk 2 above)
Comment (markaren) [reply pythonfmu/csvbuilder.py]: It has to do with headers generated from a specific tool we use.

File: pythonfmu/csvbuilder.py [hunk 4]
```diff
@@ -23,9 +25,48 @@ def lerp(v0: float, v1: float, t: float) -> float:
...
+    raise Exception("Unsupported type: " + Fmi2Type.enumeration.name)
+  else:
+    raise Exception("Could not process type from input string: " + s)
+
+TYPE2OBJ = {{
+    Fmi2Type.integer: Integer,
+    Fmi2Type.real: Real,
+    Fmi2Type.boolean: Boolean,
+    Fmi2Type.string: String
+    }}
```
[the remark this answers is not in the document]
Comment (markaren) [reply pythonfmu/csvbuilder.py]: Need double `{{}}` here

**Comment by markaren** [comment]
PR should be ready. Some stuff I did not originally meant to target for this PR was also added to make stuff work.

File: pythonfmu/fmi2slave.py [hunk 5]
```diff
@@ -146,26 +146,21 @@ def __apply_start_value(self, var: ScalarVariable):
 
         var.start = refs[0]
 
-    def register_variable(self, var: ScalarVariable):
+    def register_variable(self, var: ScalarVariable, nested: bool = True):
```
Comment (fcollonval) [inline pythonfmu/fmi2slave.py]:
```suggestion
    def register_variable(self, var: ScalarVariable, nested: bool = True):
        """Register a variable as FMU interface.
        
        Args:
            var (ScalarVariable): The variable to be registered
            nested (bool): Optional, does the "." in the variable name reflect an object hierarchy to access it? Default True
        """
```

File: pythonfmu/csvbuilder.py [hunk 6]
```diff
@@ -34,71 +75,103 @@ def __init__(self, **kwargs):
         self.current_index = 0
         self.next_index = None
         self.current_time = 0.0
+        self.interpolate = True
         data = dict()
 
         def read_csv():
             with open(self.resources + '/' + "{filename}") as f:
-                return list(csv.reader(f, delimiter=','))
+                return list(csv.reader(f, skipinitialspace=True, delimiter=',', quotechar='"'))
```
[the remark this answers is not in the document]
Comment (markaren) [reply pythonfmu/csvbuilder.py]: I feel somewhat uncomfortable with all the magic

**Comment by fcollonval** [comment]
I went ahead and merge this to finalize a PR using subcommands.

**Pull request merged**